website/static/js/filters.js

```diff
@@ -0,0 +1,149 @@
+                                    }
+                                }
+                inputs:     -a dictionary of the search fields
+                            -key: the ID for the search field
+                            -value: the selector that this field should search against
+         */
+        var settings = $.extend({
+            items: ['.items']
+        }, options);
+
+        var itemsSelector = settings.items.join();
+
+        var filterConstraints = {};
+        var searchConstraints = {};
+
+        var search = function(el) {
+            for (var key in searchConstraints) {
+                if (searchConstraints.hasOwnProperty(key)) {
+                    var content = jQuery(el.querySelector(key)).text().toLowerCase();
+                    var exists = content.indexOf(searchConstraints[key]);
+                    if (exists === -1) {
+                        return false;
+                    }
+                }
+            }
+            return true;
+        };
+
+        var filter = function(el) {
+            for (var key in filterConstraints) {
+                if (filterConstraints.hasOwnProperty(key)) {
+                    var selector = settings.groups[key].filter;
+                    var type = settings.groups[key].type;
+                    var match = filterConstraints[key];
+                    if (type === 'text') {
+                        var content = jQuery(el.querySelector(selector)).text();
+                        if (match.indexOf(content) === -1) {
+                            return false;
+                        }
+                    }
+                    else if (type === 'checkbox') {
+                        if (match.indexOf(el.querySelector(selector).checked) === -1) {
+                            return false;
+                        }
+                    }
+                }
+            }
+            return true;
+        };
+
+        var toggle = function() {
+            var activeItems = $(itemsSelector);
+            $(itemsSelector).each(function() {
+                var self = this;
+                if (!search(self)) {
+                    activeItems.splice(activeItems.index(self), 1);
+                    $(self).fadeOut();
+                }
+                else if (!filter(self)){
+                    activeItems.splice(activeItems.index(self), 1);
+                    $(self).fadeOut();
+                }
+            });
+            $(activeItems).fadeIn();
+            if (settings.callback !== undefined) {
+                var empty = [];
+                var filtered = [];
+                for (var i = 0, selector; selector = settings.items[i]; i ++) {
+                    var active = jQuery(activeItems).filter(selector);
+                    var items = $(selector);
+                    if (active.length < items.length) {
+                        filtered.push(selector);
+                    }
+                    if (active.length === 0) {
```

Comment:
It shouldn't matter.  The end result if active.length == 0 and items.length >= 1 is that filtered should then contain that selector and empty should as well and in either order this would happen.